src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -210,7 +210,7 @@ private static boolean shouldBeAloneOnLine(RightCurlyOption bracePolicy,
                 || bracePolicy == RightCurlyOption.ALONE_OR_SINGLELINE
                     && shouldBeAloneOnLineWithAloneOrSinglelineOption(details, targetSrcLine)
                 || details.shouldCheckLastRcurly
-                    && details.rcurly.getLineNo() == details.nextToken.getLineNo();
+                    && shouldBeAloneOnLineWithAloneOrSinglelineOption(details, targetSrcLine);
```

Comment:
This is slightly confusing as we are calling a `AloneOrSinglelineOption` method when the option is really `same`. If we are using this method with a new option, then the method name should be renamed.

Also this is now the same code as `alone_or_singleline` so why don't we just merge the 2 sets of code?

Reply:
Well, the description of the option `same` is also in terms of `alone-or-singleline`. First sentence of the description of `same` is

> Works like alone_or_singleline but the brace should be on the same line as the next part of a multi-block statement

So I would say it is arguable to also use that method inside the "should be alone" check that is done after the "should be same" check.

But you are right about merging the options.

Reply:
Description outside the code may mention it but it is still not clear from the code alone.
I feel we should rename to something like `shouldBeAloneOnLineWithNotAloneOption`.